torch/utils/_pytree.py

```diff
@@ -190,9 +190,10 @@ def tree_unflatten(values: List[Any], spec: TreeSpec) -> PyTree:
 
     return unflatten_fn(child_pytrees, spec.context)
 
-def tree_map(fn: Any, pytree: PyTree) -> PyTree:
+def tree_map(fn: Any, pytree: PyTree, *rest: Any) -> PyTree:
     flat_args, spec = tree_flatten(pytree)
-    return tree_unflatten([fn(i) for i in flat_args], spec)
+    all_flat_args = [flat_args] + [r for r in rest]
+    return tree_unflatten([fn(*i) for i in zip(*all_flat_args)], spec)
```

Comment:
```suggestion
    return tree_unflatten([fn(*i) for i in zip(flat_args, *rest)], spec)
```

Reply:
Nice point. Thanks for pointing it out, @Skylion007! I've updated it.